src/shipping/handlers.ts

```diff
@@ -45,12 +47,22 @@ export const createChannelsChangeHandler = (
 
 export function getCreateShippingPriceRateVariables(
   data: ShippingZoneRatesPageFormData,
-  id: string
+  id: string,
+  addPostalCodeRules: any,
+  radioInclusionType: PostalCodeRuleInclusionTypeEnum
 ): CreateShippingRateVariables {
   const parsedMinDays = parseInt(data.minDays, 10);
   const parsedMaxDays = parseInt(data.maxDays, 10);
+  const postalCodeRules = addPostalCodeRules
+    .filter(code => !code.id)
```

Comment:
In which case would they not have ids?

Reply:
when codes are newly added and not yet saved. Array which leads here contains new entries added by user and existing elements from DB (with ids). The mutation should utilize `addPostalCodeRules` only with ones with no id, if we pass it all it raises an exception.